app/src/main/java/com/example/recipeapp/adapters/RecipeSearchAdapter.java

```diff
@@ -0,0 +1,91 @@
+package com.example.recipeapp.adapters;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.navigation.Navigation;
+import androidx.navigation.fragment.NavHostFragment;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.example.recipeapp.R;
+import com.example.recipeapp.databinding.ItemRecipeCardBinding;
+import com.example.recipeapp.fragments.RecipeDetailsFragment;
+import com.example.recipeapp.models.Recipe;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RecipeSearchAdapter extends RecyclerView.Adapter<RecipeSearchAdapter.ViewHolder> {
+    private static final String TAG = "RecipeSearchAdapter";
+    private Context context;
+    public List<Recipe> recipesList;
+    private ItemRecipeCardBinding item_binding;
+
+    public RecipeSearchAdapter(Context context, List<Recipe> recipes) {
+        this.context = context;
+        this.recipesList = recipes;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        item_binding = ItemRecipeCardBinding.inflate(LayoutInflater.from(context), parent, false);
+        return new ViewHolder(item_binding);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        Recipe recipe = recipesList.get(position);
+        holder.bind(recipe);
+        Bundle bundle = new Bundle();
+        bundle.putParcelable(Recipe.class.getSimpleName(),recipe);
+        holder.itemView.setOnClickListener(Navigation.createNavigateOnClickListener(R.id.action_recipeSearchFragment_to_recipeDetailsFragment,bundle));
```

Comment:
even though we're calling methods of some of these, if we want to, we can use final. See the method signature as well.